src/librustc_driver/lib.rs

```diff
@@ -821,8 +848,15 @@ pub fn monitor<F:FnOnce()+Send+'static>(f: F) {
         cfg = cfg.stack_size(STACK_SIZE);
     }
 
-    match cfg.spawn(move || { io::set_panic(box err); f() }).unwrap().join() {
-        Ok(()) => { /* fallthrough */ }
+    match cfg.spawn(move || {
+                 io::set_panic(box err);
+                 f()
+             })
+             .unwrap()
+             .join() {
```

Comment:
I would pull the match arg out into a local variable, there is no way this is going to look nice any other way (should be `if let` too).